src/bartiq/compilation/_compile.py

```diff
@@ -388,13 +388,38 @@ def _compile(
         repetition=repetition,
         children_order=routine.children_order,
     )
-    return _add_derived_resources(compiled_routine, backend, derived_resources)
+
+    tmp_routine = (
+        compiled_routine
+        if CompilationFlags.EXPAND_RESOURCES in compilation_flags
+        else _dummify_child_resources(compiled_routine, backend)
+    )
+    tmp_routine = _add_derived_resources(tmp_routine, backend, derived_resources)
+
+    return replace(compiled_routine, resources=tmp_routine.resources)
 
 
 def _accepts_resource_name(func: Calculate[T] | CalculateWithName[T]) -> TypeIs[CalculateWithName[T]]:
     return "resource_name" in inspect.signature(func).parameters
 
 
+def _dummify_resources(compiled_routine: CompiledRoutine[T], backend: SymbolicBackend[T]) -> CompiledRoutine[T]:
```

Comment:
I don't link the word `dummify` as it's a bit meaningless on its own. Any ideas for replacement?

Reply:
While I think that `dummify` is a perfectly fine word, I also agree that we can improve the name of the functions here. I change the names to `_introduce_placeholder_resources` and `_introduce_placeholder_child_resources` respectively.